src/kernel/capinv.c

```diff
@@ -863,7 +854,8 @@ cap_introspect(struct captbl *ct, capid_t capid, u32_t op, unsigned long *retval
 	if (unlikely(!ch)) return -EINVAL;
 
 	switch(ch->type) {
-	case CAP_THD: return thd_introspect(((struct cap_thd*)ch)->t, op, retval);
+	case CAP_THD:  return thd_introspect(((struct cap_thd*)ch)->t, op, retval);
+	case CAP_TCAP: return tcap_introspect(((struct cap_tcap*)ch)->tcap, op, retval);
```

Comment:
says the person coming from PLs with required exhaustive pattern matching ;-)  I'd tend to agree, but I don't know if it is worth making this change without making a comparable change across the code-base.